x-pack/plugins/rule_registry/server/rule_data_plugin_service/resource_installer.ts

```diff
@@ -316,7 +316,7 @@ export class ResourceInstaller {
               name: ilmPolicyName,
               rollover_alias: primaryNamespacedAlias,
             },
-            'index.mapping.total_fields.limit': 1700,
+            'index.mapping.total_fields.limit': 2000,
```

Comment:
Bumping the limit here is ok as a short term fix IMO, but allowing the number of mapped fields to continuously grow at this rate is potentially risky from a performance perspective. All index mappings are stored in the global cluster state, which is [periodically shared with all nodes in the cluster](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-state-publishing.html). We should talk to the ES team to get an idea of what the performance implications of including thousands of fields in the index mapping might be. The default `total_fields.limit` value is 1000, so we're starting to move far beyond the typical default values.

If we determine that there will be significant negative performance implications from maintaining index mappings with thousands of fields, we should explore ways to reduce the number of pre-defined mappings. We might do this by splitting up the mappings into separate indices, using runtime fields to map rarely-used fields as needed, etc.

cc @rylnd @oatkiller